### PR TITLE
Hide command line instructions when merge form is visible

### PR DESCRIPTION
**Pull request by CirnoT**
Hides element introduced by https://github.com/go-gitea/gitea/pull/13840 when merge form is visible.

Before:
chrome_2020-12-12_01-36-25

After:
chrome_2020-12-12_01-37-12
chrome_2020-12-12_01-37-42

### CODE REVIEW

**Comment by CirnoT** [comment]
This is done to avoid anchor link right below the button which looks bad and can cause mis-click/tap. There is no easy way to make the element flow inline when form is visible because the neighboring element must also be inline, which form is not (and can not be).

**Comment by CirnoT** [comment]
This is a draft because `.db` applies `!important` which screws with dynamic hiding of element.

@silverwind any proposed solutions? I imagine this will crop up more and more as these helpers are being used.

**Comment by zeripath** [comment]
> This is a draft because `.dib` applies `!important` which screws with dynamic hiding of element.

How so? Is this on Firefox? - as it appears to work fine on Chrome.

Would it work better if the element was wrapped in a div that was hidden instead ?

**Comment by CirnoT** [comment]
Chromium, the `.hide()` applies `style="display: none;"` to element but `display: inline-block !important;` overrides it hence it will always be shown.

> Would it work better if the element was wrapped in a div that was hidden instead ?

Won't that screw up with inline styling, which is the whole point of adding `dib`?

File: web_src/js/index.js [hunk 1]
```diff
@@ -1136,6 +1137,7 @@ async function initRepository() {
       e.preventDefault();
       $(this).closest('.form').hide();
       $mergeButton.parent().show();
+      $('.instruct.dib').show();
```
Comment (silverwind) [inline web_src/js/index.js]: Don't select off CSS helpers please, give it a unique class instead.

File: templates/repo/issue/view_content/pull.tmpl [hunk 2]
```diff
@@ -363,8 +363,8 @@
 									{{end}}
 								</div>
 							</div>
-							<div class="dib ml-3">{{$.i18n.Tr "repo.pulls.merge_instruction_hint" | Safe}}</div>
-							<div class="instruct" style="display:none">
+							<div class="instruct dib ml-3">{{$.i18n.Tr "repo.pulls.merge_instruction_hint" | Safe}}</div>
+							<div class="instruct content" style="display:none">
```
Comment (silverwind) [inline templates/repo/issue/view_content/pull.tmpl]: `content` is too generic could lead to false CSS matches, I suggest a more unique `instructions-content` and `instructions-toggle`.

**Review by silverwind — APPROVED** [review]

**Review by zeripath — APPROVED** [review]

**Comment by lunny** [comment]
ci failed is unrelated and I have restarted it.

**Pull request merged**